src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -162,7 +163,8 @@ private function handleRaw(Request $request, int $type = self::MASTER_REQUEST)
                 if (null === $response) {
                     $msg .= ' Did you forget to add a return statement somewhere in your controller?';
                 }
-                throw new \LogicException($msg);
+
+                throw new ControllerDoesNotReturnResponseException($msg, $controller, __FILE__, __LINE__ - 17);
```

Comment:
we need a test case to ensure we won't miss updating this "17" in the future I suppose

Reply:
Tests added